File: src/2025/day06/src/test/java/ca/terrylockett/aoc2025/day06/TestDay06.java
package ca.terrylockett.aoc2025.day06;

import ca.terrylockett.aoccommon.resources.Resources;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TestDay06 {

	static final String TEST_FILE_NAME = "test-input.txt";

	static String testInput = "";

	@BeforeAll
	static void setup() {
		testInput = Resources.getInput(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	void testDay06part1() {
		assertEquals(4277556L, Day06Runner.part1(testInput));
	}

	@Test
	void testDay06part2() {
		assertEquals(3263827L, Day06Runner.part2(testInput));
	}
}

File: src/2023/day06/src/test/kotlin/ca/terrylockett/aoc2023/day06/TestDay06.kt
package ca.terrylockett.aoc2023.day06

import ca.terrylockett.aoccommon.inputfilefinder.Resources
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay06 {
	@Test
	fun part1() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(288, part1(inputFilePath))
	}

	@Test
	fun part2() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(71503, part2(inputFilePath))
	}
}
